hazelcast-sql/src/test/java/com/hazelcast/sql/impl/expression/ColumnTest.java

```diff
@@ -16,47 +16,76 @@
 
 package com.hazelcast.sql.impl.expression;
 
+import com.hazelcast.sql.SqlService;
 import com.hazelcast.sql.impl.SqlDataSerializerHook;
-import com.hazelcast.sql.impl.SqlTestSupport;
 import com.hazelcast.sql.impl.row.HeapRow;
-import com.hazelcast.sql.impl.type.QueryDataType;
 import com.hazelcast.test.HazelcastParallelClassRunner;
 import com.hazelcast.test.annotation.ParallelJVMTest;
 import com.hazelcast.test.annotation.QuickTest;
+import org.apache.calcite.rel.type.RelDataType;
 import org.junit.Test;
 import org.junit.experimental.categories.Category;
 import org.junit.runner.RunWith;
 
-import static groovy.util.GroovyTestCase.assertEquals;
+import static com.hazelcast.sql.impl.type.QueryDataType.BIGINT;
+import static com.hazelcast.sql.impl.type.QueryDataType.INT;
+import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertSame;
 
 @RunWith(HazelcastParallelClassRunner.class)
 @Category({QuickTest.class, ParallelJVMTest.class})
-public class ColumnExpressionTest extends SqlTestSupport {
+public class ColumnTest extends ExpressionTestBase {
```

Comment:
`verify` doesn't add much value here, because it generates only several invocations. Given that this is a foundational expression, we need to cover it with classical tests extensively, to ensure that the behavior is as expected. This includes, but not limited to:
1. Checking all types and their metadata, including BigInteger, Character, custom objects - we need to ensure that they are converted properly
1. Checking unexpected types - this may happen in case the plan was created for one model, but then the user loaded another model. In this case, the proper exception should be thrown from target of field extractor
1. Checking `__key` and `this` of different types - primitives, composite objects

Reply:
Added `ColumnEndToEndTest`